Tests/Form/DataTransformer/ModelsToArrayTransformerTest.php

```diff
@@ -0,0 +1,50 @@
+k<?php
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\AdminBundle\Tests\Form\DataTransformer;
+
+use Sonata\AdminBundle\Form\DataTransformer\ModelsToArrayTransformer;
+
+class ModelsToArrayTransformerTest extends \PHPUnit_Framework_TestCase
+{
+    private $modelManager;
+    private $choiceList;
+
+    protected function setUp()
+    {
+        $this->modelManager = $this->prophesize('Sonata\AdminBundle\Model\ModelManagerInterface')->reveal();
+        $this->choiceList = $this->prophesize('Sonata\AdminBundle\Form\ChoiceList\ModelChoiceList')->reveal();
```

Comment:
This one is only used in the legacy constructor, please move it there (I think it is responsible for the deprecation warnings you get, and the annotation should "cancel" them).